tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -394,30 +395,51 @@ def test_resolve_gomod_vendor_dependencies(
 
 @mock.patch("cachito.workers.pkg_managers.gomod.GoCacheTemporaryDirectory")
 @mock.patch("subprocess.run")
+@mock.patch("cachito.workers.pkg_managers.gomod.get_golang_version")
 @mock.patch("cachito.workers.pkg_managers.gomod.get_worker_config")
 @mock.patch("pathlib.Path.is_dir")
+@pytest.mark.parametrize(
+    "strict_vendor,raise_error", [(True, True), (False, False), ("default", True)]
+)
 def test_resolve_gomod_strict_mode_raise_error(
-    mock_isdir, mock_gwc, mock_run, mock_temp_dir, tmpdir
+    mock_isdir,
+    mock_gwc,
+    mock_golang_version,
+    mock_run,
+    mock_temp_dir,
+    tmpdir,
+    strict_vendor,
+    raise_error,
 ):
     mock_isdir.return_value = True
     # Mock the get_worker_config
     mock_config = mock.Mock()
-    mock_config.cachito_gomod_strict_vendor = True
+    if strict_vendor != "default":
+        mock_config.cachito_gomod_strict_vendor = strict_vendor
```

Comment:
I just realized that the `"default"` case doesn't actually test what is the default in our config, it tests that
```python
bool(mock.Mock().non_existing_property) is True
```
Maybe the case could be removed? But it's a minor detail, I don't mind it much

Reply:
Removed the 'default' case. It gave a false positive result anyway.